### PR TITLE
Keep track of version ids for outdated measures and components

### PR DESCRIPTION
**Pull request by macumber**
Keep track of version ids for outdated measures and components, remove only outdated versions when updating measures and components

Pull request overview
---------------------

 - Fixes https://github.com/NREL/OpenStudio/issues/5128

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/utilities/bcl/RemoteBCL.cpp [hunk 1]
```diff
@@ -290,13 +294,19 @@ void RemoteBCL::updateComponents() {
...
-    downloadMeasure(component.uid());
+    downloadComponent(component.uid());
     boost::optional<BCLComponent> newComponent = waitForComponentDownload();
 
     if (newComponent) {
-      boost::optional<BCLComponent> oldComponent = LocalBCL::instance().getComponent(newComponent->uid());
-      if (oldComponent && oldComponent->versionId() != newComponent->versionId()) {
-        LocalBCL::instance().removeComponent(*oldComponent);
+      // delete outdated versions of this component
+      auto it = m_outdatedComponentVersionIdMap.find(component.uid());
```
Comment (macumber) [inline src/utilities/bcl/RemoteBCL.cpp:302]: If desired, could just loop over all components here looking for the same uid and different version ids, that would save the need for the new member variables

Comment (jmarrec) [reply src/utilities/bcl/RemoteBCL.cpp:302]: I think I'd prefer that. I've used a different implementation in #5129, can you review that please?

**Comment by jmarrec** [comment]
I'm going to close it (for now at least) as superseded by #5129

**Comment by macumber** [comment]
@jmarrec good by me, I like yours better

**Pull request closed without merging**